### DISCUSSION TITLE
HELP HELP!!  (Swift) How can encode EthereumAbiFunction with fixed addresses array?

### DISCUSSION THREAD

**Discussion by DtechLabs** [trustwallet/wallet-core#2299] [Q&A]
I need encode and call this ABI function `multiRoute(address,uint256,address,address[4])`.  Last function parameter is fixed length  array of address.  
How can I do it?  May be through json Abi?

**Comment by optout21** [comment]
Currently the only fixed-length array supported is `bytesNnn`.
Could you give an address where this function is deployed (e.g. visible on etherscan)?

**Reply by DtechLabs** [reply]
main - 0xAd17d367679f0DBdf23209A70640Cff3c58f4E9
polygon - 0x70cD88f92203B1243f1Fa9c6DeD28FeF94101EFB 
and also on BSC, CEOL, Fantom, Avalance

**Reply by optout21** [reply]
thanks
I have created an issue for this (#2302).
It's not clear to me if this is compliant to the ABI format, or it is actually an address[] type, to check.
I can see `04` in the stream, which may be the array size (I would not expect that in a fixed-array case).

**Reply by DtechLabs** [reply]
Thanks.
Yes. You are right.  In case fixed size array decoder  not expected count before array. And also getType() in that case  return wrong value.  
I think need add new type of Param ->  FixedSizeArray<..>.
